### PR TITLE
Scene manager: Show statuses

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Added statuses to scene inventory manager.

## Additional info
The code of scene inventory tool was not much clear so it was "easier" to refactor some parts for easier understanding where and how data are comming in. The refactor is done only in bounds of this feature. Also discovered few bugs that probably caused issues even in current develop, if you hit a bug please make sure it is bug related to this PR.

## Screenshot
image
image

## Testing notes:
1. Launch host of your choice.
2. Load/reference some representations in.
3. Scene inventory should show them.
4. You should be able to change their version, update and remove them, switch to hero and back, switch to different representation.
5. Status should show current status of version of loaded representation.

### CODE REVIEW

File: client/ayon_core/tools/sceneinventory/select_version_dialog.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            dialog.select_index(index)
+        dialog.exec_()
+        return dialog.get_selected_item()
+
+    def _on_confirm(self):
+        self._cancelled = False
+        index = self._versions_combobox.currentIndex()
+        print(index)
+        item_id = self._versions_combobox.itemData(index, ITEM_ID_ROLE)
+        print(item_id)
```
Comment (BigRoy) [inline client/ayon_core/tools/sceneinventory/select_version_dialog.py:212]:
```suggestion
        item_id = self._versions_combobox.itemData(index, ITEM_ID_ROLE)
```

**Review by LiborBatek — CHANGES_REQUESTED** [review]
I have started testing it in ```Maya``` host app and have some first issues:

1) ```Update All``` isnt working atm

Screenshot 2024-05-30 085014


2) ```Filter outdated``` isnt working atm

3) ```Hero version``` doesnt propagate status from the origin for ie ```v003``` and ```[v003]``` doesnt match status


 
Screenshot 2024-05-30 085113

**Comment by iLLiCiTiT** [comment]
> 1. Update All isnt working atm

Fixed

> 2. Filter outdated isnt working atm

Fixed

> 3. Hero version doesnt propagate status from the origin for ie v003 and [v003] doesnt match status

Hero version has own status, it is not related to origin version.

**Review by LiborBatek — APPROVED** [review]
Now even ```Filter outdated``` and ```Update all outdated``` works ok.

I have also did some additional testing within ```Blender``` host app and all good too!

**Comment by LiborBatek** [comment]
If something to note, there is a slight UX issue with Qt formating or so...

Could be aligned to ```Version``` column imho all the time
ScnInv

**Pull request merged**